firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/BreathingCircuit/AlarmMuteService.cpp

```diff
@@ -11,9 +11,28 @@
 
 namespace Pufferfish::Driver::BreathingCircuit {
 
+void AlarmMuteService::input_clock(uint32_t current_time) {
+  if (mute_start_time_ == 0) {
+    mute_start_time_ = current_time;
+  }
+  mute_duration_ = current_time - mute_start_time_;
+}
+
 void AlarmMuteService::transform(
-    AlarmMute &alarm_mute, const AlarmMuteRequest &alarm_mute_request) {
+    uint32_t current_time, const AlarmMuteRequest &alarm_mute_request, AlarmMute &alarm_mute) {
   alarm_mute.active = alarm_mute_request.active;
+  input_clock(current_time);
+
+  if (alarm_mute.active) {
+    continue_countdown(alarm_mute);
```

Comment:
Since `continue_countdown` only has one line in it and is only used here, it might be better to merge the contents of `continue_countdown` back into here and delete the `continue_countdown` method.